spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -125,8 +120,10 @@ CloseableIterator<ColumnarBatch> open(FileScanTask task) {
     return iter.iterator();
   }
 
-  private SparkDeleteFilter deleteFilter(FileScanTask task) {
-    return task.deletes().isEmpty() ? null : new SparkDeleteFilter(task, table().schema(), expectedSchema);
+  protected SparkDeleteFilter deleteFilter(CST task) {
+    Preconditions.checkArgument(task.isFileScanTask(), "Only FileScanTask is supported for delete filtering");
```

Comment:
Is BatchDataReader only for FileScanTask?  Should we just pass this as parameter to BaseDataReader?